libvast/src/system/catalog.cpp

```diff
@@ -344,14 +344,31 @@ catalog_state::lookup_impl(const expression& expr) const {
         },
         [&](const field_extractor& lhs, const data& d) -> result_type {
           auto pred = [&](const auto& field) {
-            if (!compatible(field.type(), x.op, d))
+            auto match_name = [&] {
+              auto field_name = field.field_name();
+              auto key = std::string_view{lhs.field};
+              if (field_name.length() >= key.length()) {
+                auto pos = field_name.length() - key.length();
+                auto sub = field_name.substr(pos);
+                return sub == key && (pos == 0 || field_name[pos - 1] == '.');
+              }
+              auto layout_name = field.layout_name();
```

Comment:
there is no code path in the unit tests that exercise the case `key.length() > field_name.length()`, so it's a bit hard to see correctness.